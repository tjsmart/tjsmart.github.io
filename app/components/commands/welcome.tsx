import { Command, registerCommand } from "./factory";

const welcome = new Command(
    "welcome",
    () => {
        return (
            <>
                <h1 className="bg-orange-800 text-center">
                    {`Welcome to Tyler Smart's personal website! ðŸŽ‰`}
                </h1>
                <p>This website is inspired by modern terminal applications.</p>
                <p>
                    To get started, try running the `help` command to see what
                    commands are available.
                </p>
            </>
        );
    },
    "Welcome message displayed at on startup.",
);

registerCommand(welcome);

if (process.env["NODE_DEV"] == "TEST") {
    module.exports.welcome = welcome;
}
