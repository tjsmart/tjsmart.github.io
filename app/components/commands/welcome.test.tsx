import renderer from "react-test-renderer";
import { Command } from ".";
import * as mut from "./welcome";

// @ts-ignore
const welcome: Command = mut.welcome;

describe("welcome", () => {
    test("welcome snapshot", () => {
        const result = welcome.callback([]);
        if (typeof result == "string") {
            fail("Expected result to not to be of type string");
        }

        const component = renderer.create(result);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
