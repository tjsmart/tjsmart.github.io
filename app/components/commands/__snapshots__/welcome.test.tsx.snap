// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome welcome snapshot 1`] = `
[
  <h1
    className="bg-orange-800 text-center"
  >
    Welcome to Tyler Smart's personal website! ðŸŽ‰
  </h1>,
  <p>
    This website is inspired by modern terminal applications.
  </p>,
  <p>
    To get started, try running the \`help\` command to see what commands are available.
  </p>,
]
`;
